### PR TITLE
build: add linting for bazel build files

### PR DESCRIPTION
**Pull request by devversion**
It was always painful to create/update Bazel build files as there were
no real rules for formatting. In order to not bother too much about
formatting and having consistent `BUILD.bazel` /`.bzl` files, `buildifier`
is now running as a CI check to lint these files.

**Note**: Setup is consistent with what we have on `angular/angular`. We want to keep the buildifier logic inside of the npm scripts as we don't want to add a dependency on bash for developers. Buildifier doesn't support configuration files for the warnings.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,9 +14,12 @@
   "scripts": {
     "postinstall": "ngc -p angular-tsconfig.json",
     "build": "gulp build-release-packages",
+    "bazel:buildifier": "find . -type f \\( -name \"*.bzl\" -or -name WORKSPACE -or -name BUILD -or -name BUILD.bazel \\) ! -path \"*/node_modules/*\" | xargs buildifier -v --warnings=attr-cfg,attr-license,attr-non-empty,attr-output-default,attr-single-file,constant-glob,ctx-args,depset-iteration,depset-union,dict-concatenation,duplicated-name,filetype,git-repository,http-archive,integer-division,load,load-on-top,native-build,native-package,output-group,package-name,package-on-top,redefined-variable,repository-name,same-origin-load,string-iteration,unused-variable",
+    "bazel:format": "yarn -s bazel:buildifier --lint=fix --mode=fix",
```
Comment (mmalerba) [inline package.json]: should we change `format` to `ts:format` and make a new `format` that does both?

Comment (devversion) [reply package.json]: Wanted to do that but was afraid that most people don't need it as the Bazel buildifier goes through all build files (which is kind of slow) and linting is already slow. This is similar to what `angular/angular` does.

**Comment by jelbourn** [comment]
I'm fine with the commands being either way, could also change it in a follow-up if it's annoying and/or slow.

**Comment by devversion** [comment]
@mmalerba @jelbourn Done. FWIW: The `git-clang-format` thing still doesn't work on Windows with Git Bash 😥

**Comment by andrewseguin** [comment]
Needs rebase

**Pull request merged**